lib/src/components/workflowStep/workflowStepActions.js

```diff
@@ -6,11 +6,13 @@ export function WorkflowStepActions({ children }) {
 
   const onActionToggle = ({ type, payload }) => {
     if (type === 'ACTIVE' && !openDropdowns.includes(payload.id)) {
-      setOpenDropdowns([...openDropdowns, payload.id]);
+      setOpenDropdowns(prevOpenDropdowns => [...prevOpenDropdowns, payload.id]);
     }
 
     if (type === 'UNACTIVE') {
-      setOpenDropdowns(openDropdowns.filter(d => d !== payload.id));
+      setOpenDropdowns(prevOpenDropdowns =>
+        prevOpenDropdowns.filter(d => d !== payload.id)
+      );
```

Comment:
This was a fun race condition! I went round the houses on this but it turns out you can get the prev state with the useState hook too!

Reply:
Anytime you're working on existing state (e.g. filtering out or concating) always use the prev state as the state mechanisms are async in nature.